plugin/build.gradle.kts

```diff
@@ -69,3 +107,7 @@ tasks.jacocoTestReport {
 tasks.check {
     dependsOn(tasks.jacocoTestReport)
 }
+
+tasks.withType<Test> {
+    dependsOn("publishDefaultPublicationToPluginTestRepository")
```

Comment:
You can safely replace with:

```
tasks.withType<Test> {
    dependsOn("publishToMavenLocal")
}
```

And use MavenLocal for the whole testing scenario.

Reply:
I don't like that too much since `mavenLocal` is global to the machine and might contain other artifacts that will be pulled in tests inadvertently. Doing everything inside the repo makes sure everything is contained.

Reply:
> Doing everything inside the repo makes sure everything is contained.

I see your point, but then having this in the test gradle file would probably suffice as well:
 
```
    mavenLocal {
      content {
        includeGroup("com.yelp.codegen")
      }
    }
```

This will make sure mavenLocal is used only for `com.yelp.codegen` and nothing else.

Reply:
But then there's the other side of the thing. This will add the plugin under test plugin to mavenLocal, which a user might end up a few weeks later when wanting to pull something else from mavenLocal.

Overall, I find adding 

```
configure<PublishingExtension> {
    repositories {
        maven {
            name = "pluginTest"
            url = uri("file://${rootProject.buildDir}/localMaven")
        }
    }
}
```
is a ok price to pay for proper isolation.